src/main/java/ubc/pavlab/rdp/util/FutureUtils.java

```diff
@@ -0,0 +1,26 @@
+package ubc.pavlab.rdp.util;
+
+import org.springframework.util.concurrent.ListenableFuture;
+import org.springframework.util.concurrent.ListenableFutureCallback;
+
+import java.util.concurrent.CompletableFuture;
+
+public class FutureUtils {
+
+    public static <T> CompletableFuture<T> toCompletableFuture( ListenableFuture<T> listenableFuture ) {
+        CompletableFuture<T> cf = new CompletableFuture<T>();
+        listenableFuture.addCallback( new ListenableFutureCallback<T>() {
+
+            @Override
+            public void onSuccess( T result ) {
+                cf.complete( result );
+            }
+
+            @Override
+            public void onFailure( Throwable ex ) {
+                cf.completeExceptionally( ex );
+            }
+        } );
+        return cf;
+    }
+}
```

Comment:
This is temporary. There are alternative APIs in Spring 5 that we can migrate to.

Reply:
Hmm, there's this [WebClient](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/reactive/function/client/WebClient.html), but it's part of Spring Web Reactive and it's a much bigger change that we can handle for this project.